packages/strapi-utils/lib/provider-factory.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+
+const { createAsyncSeriesHook, createAsyncParallelHook } = require('./hooks');
+
+/**
+ * @typedef ProviderHooksMap
+ * @property willRegister
+ * @property didRegister
+ * @property willDelete
+ * @property didDelete
+ */
+
+/**
+ * Creates a new object containing various hooks used by the providers
+ * @return {ProviderHooksMap}
+ */
+const createProviderHooksMap = () => ({
+  // Register events
+  willRegister: createAsyncSeriesHook(),
```

Comment:
do you need a will / did for those events ? looks like it won't make a difference in the code

Reply:
The will event was more to notify handlers that they can update the payload & then the did event was here to say "here's the final immutable form of the data". (I kinda missed the immutable part of the did event :eyes: )